spec/lib/request_button_spec.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+require 'request_button'
+
+describe RequestButton do
+  # Used to test the request button based on a specific version, URL, and accept header.
+  #
+  # Example:
+  # context 'when version is :version' do
+  #   include_examples(
+  #     'generates the button HTML based off of the provided parameters',
+  #     :version,
+  #     version_url,
+  #     version_header
+  #   )
+  # end
+  #
+  shared_examples 'generates the button HTML based off of the provided parameters' do |version, url, header|
+    subject(:get) { RequestButton.get(version, endpoint, example_status, example_json) }
+
+    let(:expected_html) do
+      "<div><button id=\"#{button_id}\" class=\"request-button\" type=\"button\" data-status=\"#{example_status}\" "\
+      "onclick=\"#{onclick_button}\">Make Request</button>"\
+      "<a class=\"example-response-toggle hide\" onclick=\"#{onclick_anchor}\">Show Example Response</a></div>\n"\
+      "<div><h4>Example Response</h4>#{status}#{response}#{disclaimer}</div>"
+    end
+    let(:button_id) { SecureRandom.uuid }
```

Comment:
Change to something hard-coded.

Reply:
Updated here: 3c9c74f5259307fe3551eb18187a86bd30570afe